samcli/cli/types.py

```diff
@@ -54,7 +54,7 @@ def _unquote_wrapped_quotes(value):
     Unquoted string
     """
 
-    if value and ((value[0] == value[-1] == '"') or (value[0] == value[-1] == "'")):
```

Comment:
Does this change change anything since a string with mixed quotes like `'something"` now matches this?

Reply:
good catch, we need to add that `value[0] == value[-1]` to the condition.

Reply:
updated